cpp/src/parquet/file_reader.cc

```diff
@@ -536,6 +577,14 @@ std::shared_ptr<RowGroupReader> ParquetFileReader::RowGroup(int i) {
   return contents_->GetRowGroup(i);
 }
 
+void ParquetFileReader::PreBuffer(const std::vector<int>& row_groups,
+                                  const std::vector<int>& column_indices,
+                                  const ::arrow::io::CacheOptions& options) {
+  // Access private methods here
+  SerializedFile* file = static_cast<SerializedFile*>(contents_.get());
```

Comment:
`checked_cast` here. Is it possible for `Contents` to be something else than `SerializedFile`?

Reply:
It's not possible, but I made it a checked cast regardless.

Reply:
Interesting. Do we use the polymorphism only for tests?

Reply:
I actually don't think it's even used in tests. SerializedFile is the only implementation.